src/time-range-picker/combobox-picker/combobox-picker.tsx

```diff
@@ -0,0 +1,389 @@
+            endInputRef.current.select();
+        }
+
+        if (triggerOnBlur) {
+            setActiveTimeSelector(null);
+            onBlur?.();
+        }
+
+        setInitialStartTimeVal(start);
+        setInitialEndTimeVal(end);
+    };
+
+    const handleClear = (event: React.MouseEvent) => {
+        event.stopPropagation();
+        setStartTimeVal("");
+        setEndTimeVal("");
+        setInitialStartTimeVal("");
+        setInitialEndTimeVal("");
+
+        const timeValue: TimeRangePickerValue = {
+            start: "",
+            end: "",
+        };
+
+        onChange?.(timeValue);
+        setActiveTimeSelector(null);
+    };
+
+    // =============================================================================
+    // RENDER FUNCTIONS
+    // =============================================================================
+    const renderClearButton = () => {
+        if (
+            !readOnly &&
+            !disabled &&
+            (startTimeVal?.length > 0 || endTimeVal?.length > 0)
+        ) {
+            return (
+                <ClearIconContainer
+                    onClick={handleClear}
+                    type="button"
+                    aria-label="Clear"
+                >
+                    <ClearIcon aria-hidden />
+                </ClearIconContainer>
+            );
+        }
+    };
+
+    const renderDropdown = () => {
+        if (!dropdownOpen) return;
+
+        if (activeTimeSelector === "start") {
+            return (
+                <DropdownList
+                    listItems={startOptions}
+                    onSelectItem={handleStartTime}
+                    selectedItems={[startTimeVal]}
+                    disableItemFocus={true}
+                    topScrollItem={TimeHelper.findClosestFlooredTime(
+                        parseInput(startTimeVal),
+                        startOptions
+                    )}
+                />
+            );
+        } else {
+            return (
+                <DropdownList
+                    listItems={endOptions}
+                    onSelectItem={handleEndTime}
+                    selectedItems={[endTimeVal]}
+                    disableItemFocus={true}
+                    topScrollItem={TimeHelper.findClosestFlooredTime(
+                        parseInput(endTimeVal),
+                        endOptions
+                    )}
+                />
+            );
+        }
+    };
+
+    const renderElement = () => (
+        <TimeContainer
+            ref={nodeRef}
+            $disabled={disabled}
+            $error={error || !!validationError}
+            $readOnly={readOnly}
+        >
+            <RangeInputInnerContainer
+                error={error || !!validationError}
+                currentActive={
+                    activeTimeSelector === null ? "none" : activeTimeSelector
+                }
+            >
+                {/* From */}
+                <SelectorInput
```

Comment:
do include `aria-label` on the inputs to clearly indicate the purpose of each field